tests/test_samples.py

```diff
@@ -508,6 +508,16 @@ def test_beta():
         assert not numpy.array_equal(pc['weight'], weight)
 
 
+def test_beta_with_logL_infinities():
+    ns = NestedSamples(root="./tests/example_data/pc")
+    for i in range(10):
+        ns['logL'][i] = -numpy.inf
+    prior = ns.set_beta(0)
+    assert numpy.all(prior.logL[:10] == -numpy.inf)
+    assert numpy.all(prior.weight[:10] == 0)
```

Comment:
Do you need to slice to 10 for the weights? shouldn't all the weights be zero if beta is zero?

Reply:
No, should they be? All weights zero? That would be like having no samples, wouldn't it?